components/script/dom/messageevent.rs

```diff
@@ -219,7 +238,13 @@ impl MessageEventMethods for MessageEvent {
 
     /// <https://html.spec.whatwg.org/multipage/#dom-messageevent-ports>
     fn Ports(&self, cx: JSContext) -> JSVal {
-        message_ports_to_frozen_array(self.ports.borrow().as_slice(), cx)
+        let ports: Vec<DomRoot<MessagePort>> = self
```

Comment:
It's actually worth caching this in a `MutNullableDom<Heap>` or something. The spec doesn't say it's SameObject, but it probably should, and even if it doesn't we can still do it.

Reply:
Doesn't have to be this PR

Reply:
Ok, I will look into it.